frontends/ol-components/src/components/LearningResourceExpanded/InfoSectionV2.tsx

```diff
@@ -145,6 +155,77 @@ const InfoItemValue: React.FC<InfoItemValueProps> = ({
   )
 }
 
+const RunDates: React.FC<{ resource: LearningResource }> = ({ resource }) => {
+  const [showingMore, setShowingMore] = useState(false)
+  const asTaughtIn = showStartAnytime(resource)
+  const sortedDates =
+    resource.runs
+      ?.sort((a, b) => {
+        if (a?.start_date && b?.start_date) {
+          return Date.parse(a.start_date) - Date.parse(b.start_date)
+        }
+        return 0
+      })
+      .map((run) => formatRunDate(run, asTaughtIn)) ?? []
+  const showMore = allRunsAreIdentical(resource) && sortedDates.length > 2
+  if (showMore) {
+    const showMoreLink = (
+      <NoWrap>
+        <ShowMoreLink
+          color="red"
+          size="small"
+          onClick={() => setShowingMore(!showingMore)}
+        >
+          {showingMore ? "Show less" : "Show more"}
+        </ShowMoreLink>
+      </NoWrap>
+    )
+    return (
+      <span data-testid="drawer-run-dates">
+        {sortedDates.slice(0, 2).map((runDate, index) => {
+          return (
+            <NoWrap key={`run-${index}`}>
+              <InfoItemValue
+                label={runDate}
+                index={index}
+                total={showingMore ? 3 : 2}
+              />
+            </NoWrap>
+          )
+        })}
+        {!showingMore && showMoreLink}
+        {showingMore &&
+          sortedDates.slice(2).map((runDate, index) => {
+            return (
+              <NoWrap key={`run-${index + 2}`}>
+                <InfoItemValue
+                  label={runDate}
+                  index={index}
+                  total={sortedDates.length - 2}
+                />
+              </NoWrap>
+            )
+          })}
+        {showingMore && showMoreLink}
+      </span>
+    )
+  } else {
+    const runDates =
+      sortedDates.map((runDate, index) => {
+        return (
+          <NoWrap key={`run-${index}`}>
+            <InfoItemValue
+              label={runDate}
+              index={index}
+              total={sortedDates.length}
+            />
+          </NoWrap>
+        )
+      }) ?? []
```

Comment:
Left hand side will never be null or undefined, so we can remove `?? []`, although it does appear `sortedDates` can possibly be undefined from L163.